### PR TITLE
Allow partial auto-fill of review-optional page #173711737

### PR DESCRIPTION
**Pull request by akegan**
[Pivotal ticket](https://www.pivotaltracker.com/story/show/173711737)

This PR:
1. Stops hiding the review-optional page on autofill if the applicant had previously filled out the values
1. Stops resetting the review-optional page if the applicant had only partially filled out values
1. Autofills values on the review-optional page based on the previous application that we're autofilling with
1. Resets referral data on autofill if the autofill application had more than one selection for that field
1. Shows review-optional page values on the autofill preview and application review page. 

How it looks on review page:
image

How it looks on the autofill start page: 
image

How it looks if there are no values provided:
image

### CODE REVIEW

**Comment by akegan** [comment]
@james-wills-sf I just went to add testing for this PR and I couldn't find any tests for the autofill workflow, other than the [jasmine tests for _autofillReset](https://github.com/SFDigitalServices/sf-dahlia-web/pull/1385/files#diff-b85994e1a0f7365edb06aebe60c8f310L579)

I'm not sure it makes sense to add that e2e test now with the rewrite around the corner, but what do you think?

**Comment by james-wills-sf** [comment]
@akegan I do think tests are useful here but not worth a lot of time. If it would be like ~30m work I'd say it's worth it? Otherwise forget it

**Comment by akegan** [comment]
I ended up adding the skeleton for an autofill test at least, checking that review-optional stuff is filled out. That way if we touch anything else related to autofill it'll be easier to add more testing. Plus I think we will rely on our e2e tests to create the narrative of stuff that we have to reproduce on the rewrite.

**Comment by akegan** [comment]
@james-wills-sf this should be ready for review. Review instructions are a bit long because we also folded in showing the values on the autofill and review-summary pages. Let me know if you have any questions!

File: spec/e2e/pages/short-form/demographic-survey.coffee [hunk 1]
```diff
@@ -33,8 +35,8 @@ class DemographicSurvey extends AngularPage
   fill: (opts = {}) ->
     @userGender.sendKeys(@defaults.userGender)
     @genderOther.clear().sendKeys(@defaults.genderOther)
-    # Don't fill in sexual orientation so this form isn't skipped
-    # TODO: when making change to never skip demographic survey, add sexual orientation changes back.
+    @userSex.sendKeys(@defaults.userSex)
```
Comment (james-wills-sf) [inline spec/e2e/pages/short-form/demographic-survey.coffee:38]: I think you need to add "userSex" and "userSexOther" to the `fields` array in `expectToMatch` too. Hard to tell from the diff but I removed them in my previous PR: https://github.com/SFDigitalServices/sf-dahlia-web/pull/1381/files#diff-8473ca875d6ef1e7536aaadac4eef866L42

File: app/assets/javascripts/short-form/ShortFormDataService.js.coffee [hunk 2]
```diff
@@ -629,15 +614,6 @@ ShortFormDataService = (ListingDataService, ListingConstantsService, ListingPref
   # Helper functions
   #############################################
 
-  Service.checkSurveyComplete = (applicant, opts = {}) ->
```
Comment (james-wills-sf) [inline app/assets/javascripts/short-form/ShortFormDataService.js.coffee:632]: is all of this optional now?

File: spec/javascripts/services/short_form_data_service_spec.coffee [hunk 3]
```diff
@@ -1,4 +1,4 @@
-do ->
+ do ->
```
Comment (james-wills-sf) [inline spec/javascripts/services/short_form_data_service_spec.coffee:1]: is this indentation correct? Looks like it's got one single space before `do` here

**Comment by akegan** [comment]
I wasn't sure what we did in other cases, but it does look like we generally hide them if they're empty. I'll make that change.

**Comment by akegan** [comment]
@james-wills-sf OK I made the update to hide the labels if there are no values, and show "None" if there are no values at all. This is ready for re-review if you want to check it out

**Comment by james-wills-sf** [comment]
@akegan lgtm!
